ftl-build/alpine/Dockerfile

```diff
@@ -91,16 +92,17 @@ ENV BATS=/bats-core/bin/bats
 FROM builder AS tester
 
 # For FTL test compilation
-ARG CI_ARCH="x86_64-musl"
+ARG TARGETPLATFORM
+ARG TARGETARCH
+ARG TARGETVARIANT
+ARG CI_ARCH="$TARGETPLATFORM"
 ARG GIT_TAG="test-build"
 ARG GIT_BRANCH="special/CI_development"
 
 # Test compile FTL's development branch, the result is removed from the final container
-# We accept errors in the arwch_test step to be able to update the arch_tests if a change is intended
 # Run the full test suite to ensure that the container is still capable of running the tests
-RUN git clone https://github.com/pi-hole/FTL.git \
+RUN git clone https://github.com/pi-hole/FTL.git --branch "${GIT_BRANCH}" \
```

Comment:
This will checkout sources for every platform you're building which is not optimal. You can just checkout against the build platform in a dedicated stage and mount the sources here. And for cache optim you should also init and checkout instead of cloning like: https://github.com/crazy-max/docker-fail2ban/blob/c779f371005948587ed90fcc28cfd819033a8083/Dockerfile#L6-L11